android/app/src/main/java/org/fossasia/openevent/activities/SearchActivity.java

```diff
@@ -127,25 +121,28 @@ public void handleIntent(Intent intent) {
         searchQuery(query);
     }
 
-    private void searchQuery(String constraint) {
-        results.clear();
-
-        if (!TextUtils.isEmpty(constraint)) {
-            String query = constraint.toLowerCase(Locale.getDefault());
-            String wildcardQuery = String.format("*%s*", query);
-            addResultsFromTracks(wildcardQuery);
-            addResultFromSessions(wildcardQuery);
-            addResultsFromSpeakers(wildcardQuery);
-            addResultsFromLocations(wildcardQuery);
+    @Override
+    public void onSaveInstanceState(Bundle bundle) {
+        if (searchViewModel != null) {
+            searchViewModel.setSearchText(searchText);
         }
+        super.onSaveInstanceState(bundle);
     }
 
-    @Override
-    protected void onSaveInstanceState(Bundle outState) {
-        if (searchView != null) {
-            outState.putString(SEARCH, searchText);
+    private void searchQuery(String constraint) {
+        searchViewModel.setSearchText(constraint);
+        searchText = searchViewModel.getSearchText();
+
+        if (!TextUtils.isEmpty(constraint)) {
```

Comment:
The ifs and above two lines can be inside ViewModel, there is no need to just update the search text this way

Reply:
Also resolve conflicts

Reply:
@iamareebjamal I am setting the search text in the viewmodel as constraint. Why should it be inside the viewmodel ?

Reply:
@iamareebjamal And in the next line I am receiving the search text from the ViewModel and as it is a global variable we are using this search text in the `onSaveInstanceState` method, kind of storing it.

Reply:
@iamareebjamal Like I wont receive any results while observing the model until and unless the 'constraint' query is stored in the ViewModel itself !.

Reply:
You can pass the searchText in the method parameter and add the condition in ViewModel instead of activity. Setting the search text and then getting it is not appropriate. I never said about not saving it in the fragment.

About the search being done through the text, that's why I'm saying, when you pass it in the method then set it in the viewmodel itself -> Why is there a need of explicit setting of search text in view model, when it can be done from inside the viewmodel when it is received as an argument

Reply:
@iamareebjamal Is the `getSearchText` method redundant then?

Reply:
@dreadpool2 It's not redundant. You have to use it when the Activity is getting destroyed.

Reply:
@iamareebjamal I beg your pardon but I didn't understand the second part of your answer :sweat_smile: , like how the search text is obtained inside the viewmodel.

Reply:
@dreadpool2 What he means to say is, everytime you want to filter, just pass in the string as an argument to the filter method and update the string inside the ViewModel in that method only. No need for the public method ```setSearchText()```